day_7/high_level.rb

```diff
@@ -0,0 +1,30 @@
+
+class CaesarCipher
+    def encode(phrase, shift_amount)
+
+      #                65 66 67                     90    97
+      # ASCII -> [..., A, B, C, D, E, ..., X, Y, Z, ..., "a", "b"]  Location of these strings in the ASCII array.
+
+                                   #   0    1        25
+      upper_case = ("A".."Z").to_a # ["A", "B", ..., "Z"]
+      lower_case = ("a".."z").to_a # ["a"]
+      output = ""
+      phrase.each_char do |c|
+        if c == " "
+          output = output + " "
+        else
+          if c.upcase == c
+            output = output + upper_case[(c.ord - 65 + shift_amount) % 26]  #c.ord gets you the position of the character in the ascii array, subtracting 65 takes the "A" position and converts it into the position in the upper_case array. Modulo limits the range of outputs to the number of letters in the alphabet, so the code will "wrap around" if I make the shift longer, for example.
+          elsif c.downcase == c
+            output = output + lower_case[(c.ord - 97 + shift_amount) % 26]
+          end
+        end
+      end
```

Comment:
This is worth some examination- I'm sure you have, but spending some time really analyzing what code block each `end` closes may help you keep organized... why does your return output need to be where it is? If you don't know the answer, that is ok, but a question to hold onto as you learn more about context and scope.